### ISSUE TITLE
Obtain the path of the model results

### ISSUE THREAD

**Issue by teresababionunez** [ultralytics/ultralytics#14873]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi!

I want to obtain the path where my yolo model saves all the results ( the training images, results.csv, ...)

I know for example that  if I do 

metrics = yolo_model.val()
print(metrics.save_dir)

I obtain the path of the results of the validation but I want to obtain the path that it is created when I train my model from the yolo_model variable.

Thank you and kind regards.

**Comment by pderrenger** [comment]
@teresababionunez hello,

To obtain the path where your YOLO model saves all the training results, you can access the `save_dir` attribute from the training results. When you train your model, the `train` method returns a results object that includes this information. You can retrieve it similarly to how you do for validation.

For example:
```python
results = yolo_model.train(data='your_data.yaml', epochs=100)
print(results.save_dir)
```

This will give you the directory path where all training-related files are saved.

**Comment by teresababionunez** [comment]
Thank you so much for you answer!

Have a good day!

**Comment by pderrenger** [comment]
You're welcome! If you have any more questions or need further assistance, feel free to ask. Have a great day!

**Comment by teresababionunez** [comment]
Well, I have one more question. When best.pt is created, by defaults it is chosen the weights of the epoch with the best MAP-50-95 on the validation set. In results.csv, which of the metrics is this value? metrics/mAP50-95(M) or metrics/mAP50-95(B)? I am using "yolov8m-seg.pt" as pre-trained model.


Thank you again and kind regards,

Teresa Babío Núñez.

**Comment by pderrenger** [comment]
Hi Teresa,

The metric used to select the best model weights is indeed the weighted metric of mAP50 * 0.9 + mAP50-95 * 0.1. In the `results.csv` file, this value corresponds to the (M) metrics, which represent the macro average. This calculation averages the performance metric over each class, aligning with your understanding.

If you have any further questions, feel free to ask.

**Comment by teresababionunez** [comment]
Thank you so much again for the help.


Have a lovely day!

Kind regards,

Teresa Babío Núñez.

**Comment by pderrenger** [comment]
@teresababionunez you're welcome, Teresa! I'm glad I could help. If you have any more questions or need further assistance, feel free to reach out. Have a lovely day!
